x-pack/plugins/observability_solution/inventory/server/routes/entities/get_latest_entities_alerts.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery, termQuery } from '@kbn/observability-plugin/server';
+import { ALERT_STATUS, ALERT_STATUS_ACTIVE } from '@kbn/rule-data-utils';
+import { AlertsClient } from '../../lib/create_alerts_client.ts/create_alerts_client';
+import { getGroupByTermsAgg } from './get_group_by_terms_agg';
+import { IdentityFieldsPerEntityType } from './get_identity_fields_per_entity_type';
+import { EntityType } from '../../../common/entities';
+
+interface Bucket {
+  key: Record<string, any>;
+  doc_count: number;
+}
+
+type EntityTypeBucketsAggregation = Record<EntityType, { buckets: Bucket[] }>;
+
+export async function getLatestEntitiesAlerts({
+  alertsClient,
+  kuery,
+  identityFieldsPerEntityType,
+}: {
+  alertsClient: AlertsClient;
+  kuery?: string;
+  identityFieldsPerEntityType: IdentityFieldsPerEntityType;
+}): Promise<Array<{ [key: string]: any; alertsCount: number; type: EntityType }>> {
+  if (identityFieldsPerEntityType.size === 0) {
+    return [];
+  }
+
+  const filter = {
+    size: 0,
+    track_total_hits: false,
+    query: {
+      bool: {
+        filter: [...termQuery(ALERT_STATUS, ALERT_STATUS_ACTIVE), ...kqlQuery(kuery)],
+      },
+    },
+  };
+
+  const response = await alertsClient.search({
+    ...filter,
+    aggs: getGroupByTermsAgg(identityFieldsPerEntityType),
+  });
+
+  const aggregations = response.aggregations as EntityTypeBucketsAggregation;
```

Comment:
Can you do it like? 🤔

```suggestion
  const response = await alertsClient.search<EntityTypeBucketsAggregation>({
    ...filter,
    aggs: getGroupByTermsAgg(identityFieldsPerEntityType),
  });

  const aggregations = response.aggregations;
```

Reply:
I noticed that alertsClient.find accepts a generic type for aggregation, so I was expecting it to work. However, I couldn't get it to function, and my poor mac isn't helping at all. If you think it's blocking, I could take another look, but can we proceed with the current solution and revisit it later?